File: entities/src/main/java/org/odk/collect/entities/javarosa/parse/EntityXFormParserFactory.java
package org.odk.collect.entities.javarosa.parse;

import org.javarosa.xform.parse.IXFormParserFactory;
import org.javarosa.xform.parse.XFormParser;
import org.jetbrains.annotations.NotNull;

public class EntityXFormParserFactory extends IXFormParserFactory.Wrapper {

    public EntityXFormParserFactory(IXFormParserFactory base) {
        super(base);
    }

    @Override
    public XFormParser apply(@NotNull XFormParser parser) {
        EntityFormParseProcessor processor = new EntityFormParseProcessor();
        parser.addProcessor(processor);

        return parser;
    }
}

File: entities/src/main/java/org/odk/collect/entities/javarosa/parse/EntityXFormParserFactory.kt
package org.odk.collect.entities.javarosa.parse

import org.javarosa.xform.parse.IXFormParserFactory
import org.javarosa.xform.parse.XFormParser

class EntityXFormParserFactory(
    base: IXFormParserFactory,
    private val v2025enabled: () -> Boolean
) : IXFormParserFactory.Wrapper(base) {
    override fun apply(parser: XFormParser): XFormParser {
        val processor = EntityFormParseProcessor(v2025enabled)
        parser.addProcessor(processor)

        return parser
    }
}
